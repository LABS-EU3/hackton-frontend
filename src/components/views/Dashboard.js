import React from "react";
import Onboarding from "../templates/Onboarding";

const Dashboard = () => (
  <Onboarding 
    user="Mildred Pascal"
  />
);

export default Dashboard;
