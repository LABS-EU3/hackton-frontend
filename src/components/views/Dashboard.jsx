import React from "react";
import { useDispatch } from "react-redux";
import { EventOnboarding } from "../templates";
import { fetchAllEvents } from "../../store/events/actions";

const Dashboard = () => {
  const dispatch = useDispatch();

  React.useEffect(() => {
    dispatch(fetchAllEvents());
  }, [dispatch]);

  return <EventOnboarding />;
};

export default Dashboard;
