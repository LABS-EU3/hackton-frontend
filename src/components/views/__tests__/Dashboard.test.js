import React from 'react'
import { Router, Route } from 'react-router';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react'
import Dashboard from '../Dashboard';
const history = createMemoryHistory()

const renderWithRouter = Component => render(
  <Router history={history}>
    <Route component={Component} />
  </Router>
)

describe("It should render dashboard page correctly", () => {
  
    it("renders the Dashboard component correctly", () => {
      const render = () => renderWithRouter(Dashboard)
      expect(render).toMatchSnapshot();
    });
  });