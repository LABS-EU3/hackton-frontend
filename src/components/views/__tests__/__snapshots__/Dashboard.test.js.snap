// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render dashboard page correctly renders the Dashboard component correctly 1`] = `[Function]`;
